app/javascript/selectors/answersSelector.js

```diff
@@ -66,3 +66,45 @@ export function numUnanswered(buildingId, state) {
   }
   return unanswered;
 }
+
+export function numAnswered(buildingId, state) {
+  let answered = 0;
+  let answers = state.buildings[buildingId].answers;
+  let buildingQuestions = state.buildings[buildingId].questions;
+
+  for (let key in answers) {
+    let questionId = answers[key].question_id;
+    if (buildingQuestions.includes(String(questionId)) && ((answers[key].text) || (answers[key].delegation_email))) {
+      answered += 1;
+    }
+  }
+  return answered;
+}
+
+
+export function questionDataPerCategory(buildingId, categoriesArray, state) {
+  let catData = {};
+  let buildingQuestions = state.buildings[buildingId].questions;
+
+  for (let i = 0; i < categoriesArray.length; i++) {
```

Comment:
Can you rewrite this so that instead of nested for loops, you iterate a single time over all the answers, and for each answer just update the corresponding entry on `catData` appropriately? This doesn't make much of a difference at the moment, but in the future when they want to add a ton of questions and categories, performance is really important!